### PR TITLE
[AIRFLOW-3066] Adding support for AWS Batch parameters

### PR DESCRIPTION
**Pull request by hugoprudente**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-3066) issues and references them in the PR "\[AIRFLOW-3066\] Add job parameters to AWSbatch Operator"

### Description

- [x] Added the parameters for using dynamic commands on Job Execution

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
```
nosetests -v tests/contrib/operators/test_awsbatch_operator.py
test_check_success_task_not_raises (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator) ... ok
test_check_success_tasks_raises (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator) ... ok
test_check_success_tasks_raises_failed (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator) ... ok
test_check_success_tasks_raises_multiple (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator) ... ok
test_check_success_tasks_raises_pending (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator) ... ok
test_execute_with_failures (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator) ... ok
test_execute_without_failures (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator) ... ok
test_init (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator) ... ok
test_template_fields_overrides (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator) ... ok
test_wait_end_tasks (tests.contrib.operators.test_awsbatch_operator.TestAWSBatchOperator) ... ok

----------------------------------------------------------------------
Ran 10 tests in 0.146s

OK
[2018-11-24 00:59:23,032] {settings.py:203} DEBUG - Disposing DB connection pool (PID 12221)
```

### Commits

- [x] My commits all reference Jira issues: [AIRFLOW-3066] Adding support for AWS Batch parameters

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/example_dags/example_awsbatch_operator.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                "Ref::input"
+            ]
+        },
+        parameters={
+            "input": "Airflow2000"
+        },
+        queue='airflow',
+        dag=dag)
+
+    t0.set_upstream(t1)
```
Comment (Fokko) [inline airflow/contrib/example_dags/example_awsbatch_operator.py]: Can we use the `>>` notation here?

File: airflow/contrib/operators/awsbatch_operator.py [hunk 2]
```diff
@@ -94,11 +100,21 @@ def execute(self, context):
         )
 
         try:
-            response = self.client.submit_job(
-                jobName=self.job_name,
-                jobQueue=self.job_queue,
-                jobDefinition=self.job_definition,
-                containerOverrides=self.overrides)
+            if self.parameters is None:
```
Comment (Fokko) [inline airflow/contrib/operators/awsbatch_operator.py]: Do we need the `if` construction here? Can't we just pass None parameters? Or do `parameters =self.parameters or {}`

**Review by Fokko — CHANGES_REQUESTED** [review]

File: airflow/contrib/operators/awsbatch_operator.py [hunk 3]
```diff
@@ -74,6 +79,7 @@ def __init__(self, job_name, job_definition, job_queue, overrides, max_retries=4
         self.job_definition = job_definition
         self.job_queue = job_queue
         self.overrides = overrides
+        self.parameters = parameters
```
Comment (esthernamrm) [inline airflow/contrib/operators/awsbatch_operator.py]: Really happy this is being added! Would it make sense to name the keyword arg `job_parameters` to better differentiate it from the `params` keyword arg?

File: airflow/contrib/example_dags/example_awsbatch_operator.py (hunk 1 above)
Comment (hugoprudente) [reply airflow/contrib/example_dags/example_awsbatch_operator.py]: Yes, I'll make the update!

File: airflow/contrib/operators/awsbatch_operator.py (hunk 2 above)
Comment (hugoprudente) [reply airflow/contrib/operators/awsbatch_operator.py]: The API does not accept None parameters or {}, both triggers KeyError exception.

**Comment by boje** [comment]
Will this be merge into master any time soon?

**Pull request closed without merging**